src/features/RecipeLibrary/hooks/useSearchLibrary.ts

```diff
@@ -0,0 +1,58 @@
+import { SEARCH_RECIPES } from "features/RecipeLibrary/data/queries";
+import { QueryResult, useQuery } from "@apollo/client";
+import type { RecipeCard } from "features/RecipeLibrary/types";
+import {
+    GetSearchLibraryQuery,
+    GetSearchLibraryQueryVariables,
+} from "__generated__/graphql";
+
+interface UseSearchLibraryQueryResult
+    extends Pick<
+        QueryResult,
+        "loading" | "error" | "refetch" | "fetchMore" | "data"
+    > {
+    isComplete: boolean;
+    endCursor: string;
+}
+
+export const useSearchLibrary = ({
+    scope,
+    query,
+}): UseSearchLibraryQueryResult => {
+    const { data, error, loading, refetch, fetchMore } = useQuery<
+        GetSearchLibraryQuery,
+        GetSearchLibraryQueryVariables
+    >(SEARCH_RECIPES, {
+        variables: {
+            query,
+            scope,
+        },
+    });
+
+    const isComplete = !data?.library?.recipes.pageInfo.hasNextPage;
+
+    // e.node
+    const recipes: RecipeCard[] =
+        data?.library?.recipes.edges.map((it) => ({
+            id: it.node.id,
+            name: it.node.name,
+            owner: it.node.owner,
+            calories: it.node.calories || null,
+            externalUrl: it.node.externalUrl || null,
+            favorite: it.node.favorite,
+            labels: it.node.labels || [],
+            photo: it.node.photo || null,
+            totalTime: it.node.totalTime || null,
+            yield: it.node.yield || null,
+        })) || [];
+
+    return {
+        loading,
+        error,
+        refetch,
+        fetchMore,
+        isComplete,
+        endCursor: data?.library?.recipes.pageInfo.endCursor,
```

Comment:
`W:` I'd inline `isComplete` too, since it isn't used anywhere else. Pulling `endCursor` into a variable would be equally consistent, but I'd inline them.

Reply:
Is the reason that inline is better because it is more transparent what is going on and being checked?

Reply:
Yeah, basically. I went looking for where else `isComplete` was used, since there's no reason to have declared it 20 lines above, if it wasn't used somewhere else. Declaring it immediately before the `return` would be similarly unambiguous that it's just for the result.